src/module.c

```diff
@@ -1397,20 +1405,32 @@ NAPI_METHOD(dcn_stop_threads) {
 
   dcn_context->loop_thread = 0;
 
+  printf("-> dcn_stop_threads\n");
+
   if (dcn_context->imap_thread && dcn_context->smtp_thread) {
+    // printf("-- before dc_interrupt_imap_idle\n");
     dc_interrupt_imap_idle(dcn_context->dc_context);
+    // printf("-- before dc_interrupt_smtp_idle\n");
     dc_interrupt_smtp_idle(dcn_context->dc_context);
+    // printf("-- before dc_interrupt_mvbox_idle\n");
     dc_interrupt_mvbox_idle(dcn_context->dc_context);
 
+    printf("-- before uv_thread_join(imap)\n");
     uv_thread_join(&dcn_context->imap_thread);
+    printf("-- before uv_thread_join(smtp)\n");
     uv_thread_join(&dcn_context->smtp_thread);
+    printf("-- before uv_thread_join(mvbox)\n");
```

Comment:
This is printed and we know `uv_thread_join()` is hanging (since the thread function is not exiting).

Reply:
I noticed that's it not only mvbox that can race, I got the same problem for the imap thread, e.g. `uv_thread_join(&dcn_context->imap_thread)` hangs.